File: graphql-todo-items-java/src/main/java/com/example/subgraphtemplatejavadgsshows/todoitems/TodoItemsService.java
package com.example.subgraphtemplatejavadgsshows.todoitems;

import java.io.IOException;
import java.util.List;

import org.springframework.stereotype.Service;

import com.example.subgraphtemplatejavadgsshows.ApiContainer;
import org.openapitools.client.models.ToDoItem;

@Service
public class TodoItemsService {

    private final ApiContainer api;

    public TodoItemsService(ApiContainer api) {
        this.api = api;
    }

    public ToDoItem fetchById(String id) throws IOException {
        return api.todoItem().getItems().stream()
            .filter(item -> item.getId().toString().equals(id))
            .findFirst()
            .orElse(null);
    }

    public List<ToDoItem> fetchByListId(String listId) throws IOException {
        return api.todoItem().getItems().stream()
            .filter(item -> item.getListId().toString().equals(listId))
            .toList();
    }

    public int fetchIncompleteCount(String listId) throws IOException {
        return (int) api.todoItem().getItems().stream()
            .filter(item -> item.getListId().toString().equals(listId) && !item.getChecked())
            .count();
    }

}

File: graphql-todo-items-java/src/main/java/com/example/subgraphtemplatejavadgsshows/todoitems/TodoItemsService.kt
package com.example.subgraphtemplatejavadgsshows.todoitems

import com.example.subgraphtemplatejavadgsshows.APiContainer
import com.example.subgraphtemplatejavadgsshows.types.TodoItem
import org.openapitools.client.apis.TodoItemControllerApi
import org.openapitools.client.models.ToDoItem
import org.springframework.stereotype.Service


@Service
class TodoItemsService(val api: APiContainer) {
    fun fetchById(id: String): ToDoItem {
        return api.todoItem().getItems().first { it.id.toString() == id }
    }

    fun fetchByListId(listId: String): List<ToDoItem> {
        return api.todoItem().getItems().filter { it.listId.toString() == listId }
    }

    fun fetchIncompleteCount(listId: String): Int {
        return api.todoItem().getItems().count { it.listId.toString() == listId && it.checked != true}
    }
}
